crates/jxl-render/src/blend.rs

```diff
@@ -333,6 +337,80 @@ pub fn patch(
     }
 }
 
+pub fn spline(
+    image_header: &Headers,
+    base_grid: &mut [SimpleGrid<f32>],
+    spline: Spline,
+    estimated_area: u64,
+) -> crate::Result<()> {
+    // Maximum total_estimated_area_reached for Level 5
+    if estimated_area
+        > (image_header.size.height * image_header.size.width + (1 << 18)).min(1 << 22) as u64
+    {
+        tracing::warn!(
+            "Large estimated_area of splines, expect slower decoding: {}",
+            estimated_area
+        );
+    }
+    // Maximum total_estimated_area_reached for Level 10
+    // if estimated_area
+    //     > (64 * (image_header.size.height * image_header.size.width) as u64 + (1u64 << 34))
+    //         .min(1u64 << 38)
+    // {
+    //     return Err(crate::Error::Frame(
+    //         jxl_frame::Error::TooLargeEstimatedArea(estimated_area),
+    //     ));
+    // }
```

Comment:
I think we can move this with `estimated_area` into `render_features`, since this is the only usage of `estimated_area`.